pkg/kotsadmupstream/upstream.go

```diff
@@ -66,6 +67,22 @@ func DownloadUpdate(appID string, update types.Update, skipPreflights bool, skip
 		}
 
 		errMsg := finalError.Error()
+
+		var kotsApplication *kotsv1beta1.Application
+		var license *kotsv1beta1.License
+		if cause, ok := errors.Cause(finalError).(upstream.IncompatibleAppError); ok {
+			isInstall := false
+			if finalSequence == nil || *finalSequence == 0 {
```

Comment:
this is not necessarily an install. finalSequence can be nil on updates if there was a failure and a version hasn't been created it.

Reply:
you can just return `isInstall` as part of `IncompatibleAppError`

Reply:
i don't think this can ever be an install? downloadupdate is only for updates